sql/api/src/main/scala/org/apache/spark/sql/types/Metadata.scala

```diff
@@ -40,11 +40,13 @@ import org.apache.spark.util.ArrayImplicits._
  * @since 1.3.0
  */
 @Stable
-sealed class Metadata private[types] (private[types] val map: Map[String, Any])
+sealed class Metadata private[types] (
+    private[types] val map: Map[String, Any],
+    @transient private[types] val runtimeMap: Map[String, Any])
```

Comment:
I am adding a runtime-only map of configs that is not serialized or exposed to the user. It will allow me to store alternative in-memory representations for certain configs. In particular, it would allow me to store SQL as well as the expression itself for default values.

Reply:
Below is an example of how it is being used:

```
val existsDefault = extractExistsDefault(default)
val (sql, expr) = extractCurrentDefault(default)
val newMetadata = new MetadataBuilder()
  .withMetadata(f.metadata)
  .putString(EXISTS_DEFAULT_COLUMN_METADATA_KEY, existsDefault)
  .putExpression(CURRENT_DEFAULT_COLUMN_METADATA_KEY, sql, expr)
  .build()
f.copy(metadata = newMetadata)
```

Reply:
This allows me to use the expression directly without generating/parsing SQL for it.